AsyncDisplayKit/ASTableView.mm

```diff
@@ -587,18 +587,46 @@ - (ASScrollDirection)scrollDirection
   } else {
     scrollVelocity = _deceleratingVelocity;
   }
+  
   ASScrollDirection scrollDirection = [self _scrollDirectionForVelocity:scrollVelocity];
   return ASScrollDirectionApplyTransform(scrollDirection, self.transform);
 }
 
-- (ASScrollDirection)_scrollDirectionForVelocity:(CGPoint)velocity
+- (ASScrollDirection)scrollableDirections
+{
+  ASScrollDirection scrollableDirection = ASScrollDirectionNone;
+  CGFloat totalContentWidth = self.contentSize.width + self.contentInset.left + self.contentInset.right;
+  CGFloat totalContentHeight = self.contentSize.height + self.contentInset.top + self.contentInset.bottom;
+  
+  if (self.alwaysBounceHorizontal || totalContentWidth > self.bounds.size.width) { // Can scroll horizontally.
+    scrollableDirection |= ASScrollDirectionHorizontalDirections;
+  }
+  if (self.alwaysBounceVertical || totalContentHeight > self.bounds.size.height) { // Can scroll vertically.
+    scrollableDirection |= ASScrollDirectionVerticalDirections;
+  }
+  return scrollableDirection;
+}
+
+- (ASScrollDirection)_scrollDirectionForVelocity:(CGPoint)scrollVelocity
 {
   ASScrollDirection direction = ASScrollDirectionNone;
-  if (velocity.y < 0.0) {
-    direction = ASScrollDirectionDown;
-  } else if (velocity.y > 0.0) {
-    direction = ASScrollDirectionUp;
+  ASScrollDirection scrollableDirections = [self scrollableDirections];
+  
+  if (ASScrollDirectionContainsHorizontalDirection(scrollableDirections)) { // Can scroll horizontally.
```

Comment:
Do we really want this behavior in table? I am not aware of a case that it can be made to scroll horizontally, at least in a scenario that we would want to trigger either a range controller or batch operation.

For example, you could transform by rotation 90º and have it scroll horizontally, but from the perspective of the table I believe it would still have only a Y value change and internally behave as though vertical

It is a great point that we can enable scroll bouncing, but even in this case I don't think we need a batch or range controller?